### ISSUE TITLE
[PLAYBACK]  Black video playback with Failed to add vertex code to shader error

### ISSUE THREAD

**Issue by malnaanah** [olive-editor/olive#1646]
I am trying to try Olive 0.2. but the video is shown as black screen and only audio is playing. Also the app crashes a lot. 
I get this error:
[ERROR] Failed to add vertex code to shader
I use kubuntu 20.04 and I have old Dell Inspiron-N5030 laptop with Intel Corporation Mobile 4 Series Chipset Integrated Graphics Controller.

Version 0.1 works fine.

Any idea??

**Comment by rodlie** [comment]
Your GPU is probably not supported. I believe you max out on OpenGL 2.1 (on Linux, even lower on Windows).

**Comment by malnaanah** [comment]
What is the minimum required OpenGL for olive

**Comment by rodlie** [comment]
I believe OpenGL 3.x is the minimum (someone correct me if I'm wrong).

**Comment by itsmattkc** [comment]
Our official policy is that your GPU is simply too old, however we have discussed downgrading to ES 2.0 before for wider support/compatibility. I'd coincidentally been working on the OpenGL code and found that downgrading was less difficult than expected and appears to work correctly on all the GPUs/platforms I have access to (Intel Integrated, AMD Integrated, NVIDIA GTX 1070, Apple M1 between Windows, macOS, and Linux). This is now merged into master with `ae3393f03141ec0f179523670950e2483b8d5c33`. This may or may not solve your problem, but this is as far as we can go without sacrificing quality. If it still doesn't work, your GPU is simply too old for what we're doing in 0.2.
